### PR TITLE
issue: 959307 Link VMA with libnl1 or libnl3 based on libibverbs

### PR DESCRIPTION
**Pull request by OphirMunk**
Signed-off-by: Ophir Munk <ophirmu@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/857/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/750/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Jenkins is failing with good  reason:
12:52:45 checking for LIBNL3... yes
12:52:45 checking for LIBNL1... no
12:52:45 configure: error: libibverbs is linked with libnl1 while libnl1-devel is not installed

**Comment by OphirMunk** [comment]
Jenkins is failing on Fedora22 and Fedora23 where libibverbs is linked with libnl1 while only libnl3 is installed.
@igor-ivanov @Di0gen - can we install updated libibverbs (with libnl3) on those machines? Any other suggestion?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/865/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/754/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
> @igor-ivanov @Di0gen - can we install updated libibverbs (with libnl3) on those machines? Any other suggestion?
@OphirMunk I think that technically it is possible but purpose of using VM in jenkins is to check original default distribution according initial discussion. @Di0gen am I correct and what do you think about?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/874/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/849/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1202/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1683/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1122/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1689/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1128/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1412/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1417/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1418/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1419/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1420/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1438/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1443/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1446/ for details (Mellanox internal link).

File: configure.ac [hunk 1]
```diff
@@ -867,20 +872,42 @@ AS_IF([test "x$have_mp_rq" == xyes],
...
+fi
+
+have_libnl1=no
+have_libnl3=no
+
+if test x"$ibverbs_libnl" == x3; then
+	PKG_CHECK_MODULES([LIBNL3],[libnl-route-3.0],have_libnl3=yes AC_DEFINE([HAVE_LIBNL3], [1], [Use libnl-route-3.0]),[:])
+	if test "$have_libnl3" == no; then
+		AC_MSG_ERROR([libibverbs is linked with libnl3 while libnl3-devel is not installed. Please install libnl3-devel and try again])	
+	fi
```
Comment (igor-ivanov) [inline configure.ac]: please consider moving this under line 879

File: configure.ac [hunk 2]
```diff
@@ -867,20 +872,42 @@ AS_IF([test "x$have_mp_rq" == xyes],
...
+else 
+	PKG_CHECK_MODULES([LIBNL1],[libnl-1], have_libnl1=yes AC_DEFINE([HAVE_LIBNL1], [1], [Use libnl-1]), [:]) 
+	if test "$have_libnl1" == no; then
+		AC_MSG_ERROR([libibverbs is linked with libnl1 while libnl1-devel is not installed. Please install libnl1-devel and try again])	
+	fi
+fi
```
Comment (igor-ivanov) [inline configure.ac]: please consider moving this under line 881

File: configure.ac [hunk 3]
```diff
@@ -867,20 +872,42 @@ AS_IF([test "x$have_mp_rq" == xyes],
...
+
+if test "$have_libnl3" = yes; then
 	AC_SUBST([LIBNL_LIBS], "$LIBNL3_LIBS")
 	AC_SUBST([LIBNL_CFLAGS], "$LIBNL3_CFLAGS")
 	AC_SUBST([LIBNLX_DEVEL], "libnl3-devel")
```
Comment (igor-ivanov) [inline configure.ac]: please consider moving it under previous if condition as
```
if test "$have_libnl3" == no; then
	AC_MSG_ERROR([AC_MSG_ERROR([libibverbs is linked with libnl3 while libnl3-devel is not installed. Please install libnl3-devel and try again])])	
elif
 	AC_SUBST([LIBNL_LIBS], "$LIBNL3_LIBS")
  	AC_SUBST([LIBNL_CFLAGS], "$LIBNL3_CFLAGS")		  	AC_SUBST([LIBNL_CFLAGS], "$LIBNL3_CFLAGS")
  	AC_SUBST([LIBNLX_DEVEL], "libnl3-devel")		  	AC_SUBST([LIBNLX_DEVEL], "libnl3-devel")
fi
```

File: configure.ac [hunk 4]
```diff
@@ -867,20 +872,42 @@ AS_IF([test "x$have_mp_rq" == xyes],
...
 else
-	PKG_CHECK_MODULES([LIBNL1],[libnl-1], have_libnl=yes AC_DEFINE([HAVE_LIBNL1], [1], [Use libnl-1]),  AC_MSG_ERROR([libnl3-devel or libnl-devel must be installed in order to build VMA]))
 	AC_SUBST([LIBNL_LIBS], "$LIBNL1_LIBS")
 	AC_SUBST([LIBNL_CFLAGS], "$LIBNL1_CFLAGS")
 	AC_SUBST([LIBNLX_DEVEL], "libnl-devel")
 fi
```
Comment (igor-ivanov) [inline configure.ac]: the same as above

File: configure.ac [hunk 5]
```diff
@@ -303,10 +303,15 @@ AC_ARG_WITH(ofed,
...
-else
+else 
         LDFLAGS="$LDFLAGS -L$ac_cv_ofed_path/lib"
 fi
 
+if test -f "$ac_cv_ofed_path/lib64/libibverbs.so" ; then
+	libibverbs_file="$ac_cv_ofed_path/lib64/libibverbs.so"
+else 
+	libibverbs_file="$ac_cv_ofed_path/lib/libibverbs.so"
+fi
```
Comment (rafiw) [inline configure.ac]: also here https://github.com/Mellanox/libvma/blob/master/configure.ac#L68

**Comment by ilansmith** [comment]
#282 has been merged.

**Pull request closed without merging**